### ISSUE TITLE
`gsutil -m` changes behavior of permissions requirements for `cp -r`/`rsync -r`

### ISSUE THREAD

**Issue by KennethLundberg** [GoogleCloudPlatform/gsutil#1663]
gsutil version 5.18

I have a service account with Storage Object Admin for a particular bucket. Sometime prior to 5.18 running `gsutil -m cp -r <src> <dst>` worked as expected (files were copied with no issue). In 5.18 I now get:
```
AccessDeniedException: 403 <sa>@<project>.iam.gserviceaccount.com does not have storage.buckets.get a
ccess to the Google Cloud Storage bucket. Permission 'storage.buckets.get' denied on resource (or it may not exist).
```
however if I run the same command using the same service account without the `-m` flag (`gsutil cp -r <src> <dst>`), the copy again succeeds.

This is my first time working with this repo, so I haven't dug into it but it appears it could be related to #1639 / #1655 which looks like it went out in 5.18.

~I haven't tested 5.17 as I'm not sure how to figure out what google-cloud-cli version gsutil 5.17 corresponds to, but happy to test that out if I can get some direction.~

Edit: Confirmed introduced in 5.18.
As others have posted below mitigations are (only one is needed):
* ([official recommendation](https://cloud.google.com/storage/docs/release-notes#January_19_2023)) Reverting to gsutils 5.17 thanks @Chupaka
* Removing `-m` flag from command
* Assigning `Storage Legacy Bucket Reader` role which grants storage.buckets.get

**Comment by steezeburger** [comment]
I was about to create my own issue. I'm experiencing something similar but maybe slightly different.

I have `$GOOGLE_APPLICATION_CREDENTIALS` set correctly, and I'm calling
`gsutil -m rsync -r -d app/static ${DJANGO_STATICFILES_URL}`

This has worked fine for a long time, but today my ci/cd starting failing with
```
ServiceException: 401 Anonymous caller does not have storage.buckets.get access to the Google Cloud Storage bucket. Permission 'storage.buckets.get' denied on resource (or it may not exist).
```

gsutil version: 5.18

**Comment by fernandobrito** [comment]
We are having the same issue in our repos. 

We use the `google-github-actions/setup-gcloud@v1` GitHub Action, which installs the latest version of `gsutil`.

`gsutil -m rsync -d -r plugins $BUCKET/plugins` was running fine until yesterday, but today it started failing with 
```
AccessDeniedException: 403 [...] does not have storage.buckets.get access to the Google Cloud Storage bucket. Permission 'storage.buckets.get' denied on resource (or it may not exist).
```

**Comment by dilipped** [comment]
Thank you everyone for reporting this issue! We are actively working on a fix.

**Comment by herbertrlee** [comment]
If you're unable to change your gsutil version for whatever reason, but are able to change your service account's IAM permissions, you can (temporarily) grant it the Storage Admin role rather than Storage Object Admin role in order to work around this issue.

**Comment by Chupaka** [comment]
The official workaround is to revert to an earlier version: https://cloud.google.com/storage/docs/release-notes#January_19_2023

@herbertrlee Storage Admin grants `storage.buckets.setIamPolicy` permission, so it's better to add "Storage Legacy Bucket Reader" - it will add the only `storage.buckets.get` permission to the Storage Object Admin's set.

**Comment by hiroshi** [comment]
FYI: gcloud 415 doesn't have fix for this issue (no update of gsutil. still 5.18).
https://cloud.google.com/sdk/docs/release-notes#41500_2023-01-24

**Comment by dilipped** [comment]
Sorry, the fix missed the 415 cut, but it will mostly likely part of Cloud SDK 416.0.0 release.

**Comment by Chupaka** [comment]
Fixed in Cloud CLI 416.0.0 and gsutil 5.19.

**Comment by dilipped** [comment]
Yes, this has been fixed in gsutil v5.19.
